DuckDuckGo/NetworkProtectionConvenienceInitialisers.swift

```diff
@@ -22,24 +22,34 @@
 import NetworkProtection
 import UIKit
 import Common
+import NetworkExtension
+
+private class DefaultTunnelSessionProvider: TunnelSessionProvider {
+    func activeSession() async -> NETunnelProviderSession? {
+        try? await ConnectionSessionUtilities.activeSession()
+    }
+}
```

Comment:
Just a simple mechanism to start injecting the session provider instead of relying on the static call directly.  This is useful to start moving in the right direction IMO.